internal/loadbalancers/delete.go

```diff
@@ -0,0 +1,85 @@
+// Copyright © 2022 Equinix Metal Developers <support@equinixlbaas.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package loadbalancers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func (c *Client) Delete() *cobra.Command {
+	var (
+		force          bool
+		loadbalancerID string
+		confirmation   string
+	)
+	deleteLoadBalancer := func(id string) error {
+		_, err := c.loadBalancerService.DeleteLoadBalancer(context.Background(), id).Execute()
+		if err != nil {
+			return err
+		}
+		fmt.Println("LoadBalancer", loadbalancerID, "successfully deleted.")
+		return nil
+	}
+	// deleteLoadBalancerCmd represents the deleteLoadBalancer command
+	deleteLoadBalancerCmd := &cobra.Command{
+		Use:   `delete --id <loadbalancer_UUID> [--force]`,
+		Short: "Deletes a loadbalancer.",
+		Long:  "Deletes the specified loadbalancer with a confirmation prompt. To skip the confirmation use --force. You can't delete a loadbalancer that has active resources. You have to deprovision all pools and ports from a loadbalancer in order to delete it.",
```

Comment:
So when a loadbalancer is deleted, it will delete the associated ports. But this is correct in that pools and origins need cleaned up separately

Reply:
Got it. Pools and origins are independent. ports do not need to be deleted in advance.